### ISSUE TITLE
cholesky keywords

### ISSUE THREAD

**Issue by andreasnoack** [FluxML/Zygote.jl#121]
It seems that Zygote understands `cholesky` only when no keyword is passed.
```
julia> gradient(s -> ones(1)'*(cholesky(reshape(s, 1, 1))\ones(1)), ones(1))
([-1.0],)

julia> gradient(s -> ones(1)'*(cholesky(reshape(s, 1, 1), check=false)\ones(1)), ones(1))
ERROR: MethodError: no method matching adjoint(::Zygote.Context, ::typeof(cholesky), ::Array{Float64,2}; check=false)
```

**Comment by MikeInnes** [comment]
Ok, interesting, so this is only because the [custom adjoint](https://github.com/FluxML/Zygote.jl/blob/ef549246918f6d013c9d38112a4479e093337f6b/src/lib/array.jl#L177) doesn't support kwargs, but this clearly could use a better error message.

Adding this keyword arg may be quite easy to do.

**Comment by willtebbutt** [comment]
Yeah, when I implemented that adjoint I didn't do a particularly thorough job. It should be fine just to add the `check` kwarg and propagate it to the call to `cholesky` inside the custom adjoint, but if you want use pivoting the implementation of the adjoint will presumably change, so that's a bit more work.

**Comment by andreasnoack** [comment]
Good point. It would probably quite more work to fix the pivoting case so initially, we should probably just allow `check` but error out if people ask for the pivoted version.
